scipy/stats/_stats.pyx

```diff
@@ -753,3 +753,87 @@ def gaussian_kernel_estimate(points, values, xi, precision, dtype, real _=0):
                 estimate[j, k] += values_[i, k] * arg
 
     return np.asarray(estimate)
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def gaussian_kernel_estimate_log(points, values, xi, precision, dtype, real _=0):
+    """
+    def gaussian_kernel_estimate_log(points, real[:, :] values, xi, precision)
+    Evaluate the log of the estimated pdf on a provided set of points.
+    Parameters
```

Comment:
Did you mean this part `def gaussian_kernel_estimate_log(points, real[:, :] values, xi, precision)`? Actually I didn't care for it but I merely copied from `gaussian_kernel_estimate`. 

I think this line could be `def gaussian_kernel_estimate_log(points, values, xi, precision)`

For reference, you can refer to https://cython.readthedocs.io/en/stable/src/userguide/language_basics.html#python-functions-vs-c-functions .

Reply:
After Cython code is compiled, some editors can't show arguments in the pop-up hint properly, but they would show like this.
`fused_cython_function: gaussian_kernel_estimate_log(*args, **kwargs)`

To show arguments, including a signature in a docstring is a better way.

In this discussion, a user showed the idea of including a signature in a docstring.
https://groups.google.com/g/pystatsmodels/c/dpVe6e0DaAM/m/FrInuMQFyB8J

The discussion was in 2013, and there may exist other solutions.

Reply:
Thanks for the reference. Since it's from Ralf (not just a user but our main maintainer I would say 😉) no problem. @tylerjereddy do we still need to do this though. The discussion is old so this might have changed. In the recent Cython I've seen been included we don't do this. But if it's a best practice, I am happy to consider this for next time.

Reply:
Some functions contain a signature in the docstrings.
Like below
https://github.com/scipy/scipy/blob/9bb9189a710130f6b00b7dcde74a83d375e1e282/scipy/spatial/_ckdtree.pyx#L671-L677
https://github.com/scipy/scipy/blob/9bb9189a710130f6b00b7dcde74a83d375e1e282/scipy/spatial/_ckdtree.pyx#L850-L856

Others don't have it.
https://github.com/scipy/scipy/blob/9bb9189a710130f6b00b7dcde74a83d375e1e282/scipy/signal/_peak_finding_utils.pyx#L20-L30

It seems that there are no specific conventions authors follow.

Reply:
@europeanplaice exactly, that's why I am asking here if we should prefer one approach over the other. Although I have added Cython in the library too, I am not a power user and it would be good to know what is the recommended approach.

Reply:
I'm also not a professional user, but in my opinion, it's ok to have a signature in docstrings if there is not a rule of it.
Although it looks a bit odd, I don't think having a signature in the docstrings reduces readability as help for users.
In my environment, the docstring is shown like this.
![image](https://user-images.githubusercontent.com/38364983/156000949-0ddf2fbd-fb33-4cc6-aab9-bada1d3a8bea.png)

Reply:
The functions/methods that do have a signature in the docstring are the ones that are directly providing public API. That's when it is needed for the html docs. In this case, it's a private function that's used from pure Python code within SciPy, so there's no point in adding the signature.

Reply:
Although, in this particular case, I think the point was to express the `real[:, :]` because that's now hidden by the fused types usage in an unusual way.

Reply:
I'll fix the spacing and keep the signature.